File: app/src/main/java/com/gk/connection/ConnectionUtil.java
package com.gk.connection;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class ConnectionUtil {

    public static boolean isNetworkAvailable(Context context){
        try {
            ConnectivityManager connectivity = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
            if (connectivity != null && connectivity.getAllNetworkInfo() != null) {
                for (NetworkInfo anInfo : connectivity.getAllNetworkInfo())
                    if (anInfo.getState() == NetworkInfo.State.CONNECTED)
                        return true;
            }
        } catch (Exception e) {
            e.printStackTrace();
            return false;
        }
        return false;
    }
}

File: app/src/main/java/com/gk/connection/ConnectionUtil.kt
package com.gk.internetreceiver

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo

open class ConnectionUtil {

    companion object {
        public fun isNetworkAvailable(context: Context): Boolean {
            try {
                val connectivity = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
                if (connectivity != null && connectivity.allNetworkInfo != null) {
                    for (anInfo in connectivity.allNetworkInfo)
                        if (anInfo.state == NetworkInfo.State.CONNECTED)
                            return true
                }
            } catch (e: Exception) {
                e.printStackTrace()
                return false
            }

            return false
        }
    }

}
